src/tests/audio/tasks/text_to_speech_test.py

```diff
@@ -3,31 +3,100 @@
     from senselab.utils.data_structures.device import DeviceType
-    from senselab.utils.data_structures.model import HFModel, SenselabModel
+    from senselab.utils.data_structures.language import Language
+    from senselab.utils.data_structures.model import HFModel, SenselabModel, TorchModel
 
     @pytest.fixture
     def hf_model() -> HFModel:
         """Fixture for the HF model."""
         return HFModel(path_or_uri="suno/bark-small", revision="main")
 
-    def test_synthesize_texts(hf_model: HFModel) -> None:
+    @pytest.fixture
+    def hf_model2() -> HFModel:
+        """Fixture for HF model."""
+        return HFModel(path_or_uri="facebook/mms-tts-eng", revision="main")
+
+    @pytest.fixture
+    def mars5_model() -> TorchModel:
+        """Fixture for MARS5 model."""
+        return TorchModel(path_or_uri="Camb-ai/mars5-tts", revision="master")
+
+    @pytest.fixture
+    def style_tts2() -> TorchModel:
+        """Fixture for StyleTTS2 model."""
+        return TorchModel(path_or_uri="wilke0818/StyleTTS2-TorchHub", revision="main")
+
+    @pytest.mark.parametrize("hf_model", ["hf_model", "hf_model2"], indirect=True)
+    def test_synthesize_texts_with_hf_model(hf_model: HFModel) -> None:
         """Test synthesizing texts."""
-        texts = ["Hello world"]
+        texts = ["Hello world", "Hello world again."]
         audios = synthesize_texts(texts=texts, model=hf_model)
 
-        assert len(audios) == 1
+        assert len(audios) == 2
         assert isinstance(audios[0], Audio)
         assert audios[0].waveform is not None
         assert audios[0].sampling_rate > 0
 
-    def test_huggingface_tts_pipeline_factory(hf_model: HFModel) -> None:
+    # TODO: create support for StyleTTS2 which currently has some dependency issues
+    # def test_synthesize_texts_with_styletts2_model(style_tts2: TorchModel, mono_audio_sample: Audio) -> None:
+    #     """Test synthesizing texts."""
+    #     texts_to_synthesize = ["Hello world", "Hello world again."]
+    #     terget_audio_resampling_rate = 24000
+    #     target_audio_ground_truth = "This is Peter."
+    #     language = Language(language_code="en")
+
+    #     resampled_mono_audio_sample = resample_audios([mono_audio_sample], terget_audio_resampling_rate)[0]
+    #     target_audio = extract_segments([(resampled_mono_audio_sample, [(0.0, 1.0)])])[0][0]
+    #     audios = synthesize_texts(
+    #         texts=texts_to_synthesize,
+    #         target=[(target_audio, target_audio_ground_truth), (target_audio, target_audio_ground_truth)],
+    #         model=style_tts2,
+    #         language=language,
+    #         force_reload=True,
+    #     )
+
+    # assert len(audios) == 2
+    # assert isinstance(audios[0], Audio)
+    # assert audios[0].waveform is not None
+    # assert audios[0].sampling_rate == terget_audio_resampling_rate
+
+    def test_synthesize_texts_with_mars5_model(mars5_model: TorchModel, mono_audio_sample: Audio) -> None:
```

Comment:
this test takes a really long time but not sure if there is anything we can do about that.

Reply:
i just think that the problem is with it running very inefficiently on the CPU....